marimo/_ast/sql_visitor.py

```diff
@@ -0,0 +1,160 @@
+
+
+def find_created_tables(sql_statement: str) -> list[str]:
+    """
+    Find the tables created in a SQL statement.
+
+    This function uses the DuckDB tokenizer to find the tables created
+    in a SQL statement. It returns a list of the table names created
+    in the statement.
+
+    Args:
+        sql_statement: The SQL statement to parse.
+
+    Returns:
+        A list of the table names created in the statement.
+    """
+    if not DependencyManager.duckdb.has():
+        return []
+
+    import duckdb
+
+    def remove_comments(sql: str) -> str:
+        # Function to replace comments with spaces, preserving newlines
+        def replace_with_spaces(match: re.Match[Any]) -> str:
+            return " " * len(match.group())
+
+        # Split the SQL into strings and non-strings
+        parts = re.split(r'(\'(?:\'\'|[^\'])*\'|"(?:""|[^"])*")', sql)
+
+        for i in range(0, len(parts), 2):
+            # Remove single-line comments
+            parts[i] = re.sub(
+                r"--.*$", replace_with_spaces, parts[i], flags=re.MULTILINE
+            )
+
+            # Remove multi-line comments
+            parts[i] = re.sub(r"/\*[\s\S]*?\*/", replace_with_spaces, parts[i])
+
+        # Join the parts back together
+        return "".join(parts)
+
+    sql_statement = remove_comments(sql_statement)
+
+    tokens = duckdb.tokenize(sql_statement)
+    created_tables: list[str] = []
+    i = 0
+
+    def token_str(i: int) -> str:
+        token = tokens[i]
+        start = token[0]
+        end = len(sql_statement) - 1
+        if i + 1 < len(tokens):
+            end = tokens[i + 1][0]
+        return sql_statement[start:end].strip()
+
+    def keyword_token_str(i: int) -> str:
+        return token_str(i).lower()
+
+    def token_type(i: int) -> str:
+        return tokens[i][1]
+
+    # See https://duckdb.org/docs/sql/statements/create_table#syntax
+    # for more information on the CREATE TABLE syntax.
+    while i < len(tokens):
+        if (
+            keyword_token_str(i) == "create"
+            and token_type(i) == duckdb.token_type.keyword
+        ):
+            i += 1
+            if i < len(tokens) and keyword_token_str(i) == "or":
+                i += 2  # Skip 'OR REPLACE'
+            if i < len(tokens) and keyword_token_str(i) in (
+                "temporary",
+                "temp",
+            ):
+                i += 1  # Skip 'TEMPORARY' or 'TEMP'
+            if i < len(tokens) and keyword_token_str(i) == "table":
+                i += 1
+                if i < len(tokens) and keyword_token_str(i) == "if":
+                    i += 3  # Skip 'IF NOT EXISTS'
+                if i < len(tokens):
+                    table_name = token_str(i)
+                    # Remove quotes if present
+                    if table_name.startswith('"') and table_name.endswith('"'):
+                        table_name = table_name[1:-1]
+
+                    created_tables.append(table_name)
```

Comment:
i think its worth preserving the case in this and then check for case insensitive later, since you may want clearer error messages with the correct case